samtranslator/model/sam_resources.py

```diff
@@ -886,6 +889,11 @@ def _construct_version(
             # If SnapStart is enabled we want to publish a new version, to have the corresponding snapshot
             if function.SnapStart and function.SnapStart.get("ApplyOn", "None") != "None":
                 logical_dict.update({"SnapStart": function.SnapStart})
+            # We can't directly change AutoPublishAlias as that would be a breaking change, so we have to add this opt-in
+            # property that when set to true would change the lambda version whenever a property in the lambda function changes
+            if self.AutoPublishAliasAllProperties:
+                properties = function._generate_resource_dict().get("Properties", {})
+                logical_dict.update(properties)
```

Comment:
Why not `logical_dict = properties`?

Reply:
Thought something like that would mess with the `code_dict` thats created and added to the `logical_dict` earlier in the file but doesn't look like it does. Going to restructure this so that `AutoPublishAliasAllProperties ` is checked first